### ISSUE TITLE
BUG: minHeight and minWidth not working

### ISSUE THREAD

**Issue by melvinvdb** [ghiscoding/Angular-Slickgrid#133]
**I'm submitting a bug report**

* **Angular Version:**
7.2.6
* **Library Version:**
2.3.3

* **Browser:**
all

* **Language:**
TypeScript 3.2.2

**Current behavior:**
minHeight and minWidth not working

**Expected/desired behavior:**
minHeight and minWidth working

* **What is the expected behavior?**
The minHeight and minWidth aren't working at all.

* **What is the motivation / use case for changing the behavior?**
Very simple bug.
See:
https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/modules/angular-slickgrid/services/resizer.service.ts
`const minHeight = (autoResizeOptions && autoResizeOptions.minHeight && autoResizeOptions.minHeight < 0) ? autoResizeOptions.minHeight : DATAGRID_MIN_HEIGHT;`
`autoResizeOptions.minHeight < 0` should be `autoResizeOptions.minHeight > 0`
and
`const minWidth = (autoResizeOptions && autoResizeOptions.minWidth && autoResizeOptions.minWidth < 0) ? autoResizeOptions.minWidth : DATAGRID_MIN_WIDTH;`
`autoResizeOptions.minWidth < 0` should also be `autoResizeOptions.minWidth > 0`

Of course minHeight and minWidth should be higher than 0 to be effective.

**Comment by ghiscoding** [comment]
Hmm this code was reviewed in multiple occasions, I see the problem now but I think the following code is better
```ts
const maxHeight = autoResizeOptions && autoResizeOptions.maxHeight && autoResizeOptions.maxHeight || undefined;
const minHeight = autoResizeOptions && autoResizeOptions.minHeight && autoResizeOptions.minHeight || DATAGRID_MIN_HEIGHT;
const maxWidth = autoResizeOptions && autoResizeOptions.maxWidth && autoResizeOptions.maxWidth || undefined;
const minWidth = autoResizeOptions && autoResizeOptions.minWidth && autoResizeOptions.minWidth || DATAGRID_MIN_WIDTH;
```

**Comment by melvinvdb** [comment]
I definitely agree!

**Comment by ghiscoding** [comment]
Ok I'll push the new code soon, I might wait couple days to release a new version though. It should available by next week. Thanks for the feedback

**Comment by melvinvdb** [comment]
Glad to help!
Do you have any idea when something like autoHeight might be available? It would be great if the grid has a possibility to fit the height to the content available (so there are no vertical scrollbars).

**Comment by ghiscoding** [comment]
I didn't even know what `autoHeight` property was for until yesterday, when I saw an [issue](https://github.com/6pac/SlickGrid/pull/342) with it in the core lib (which I also contribute a lot to). So to answer your question, I see that I'm missing the `autoHeight` option in the [GridOption interface](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/src/app/modules/angular-slickgrid/models/gridOption.interface.ts), I'll simply add it and it should work. Technically speaking all Slickgrid options should be available in that interface, if there's any that I'm missing (I thought I did them all already), we can just add them to the interface. Good catch 👍
